torch/_dynamo/utils.py

```diff
@@ -3355,6 +3355,29 @@ def wrapper(*args):
     return wrapper
 
 
+@dataclasses.dataclass
+class CompiledAutogradMetrics:
+    id: int
+    start_time_ns: int
+    end_time_ns: int
+
+
+def get_compiled_autograd_metrics(maybe_gm):
+    if (
+        not isinstance(maybe_gm, torch.fx.GraphModule)
+        or not hasattr(maybe_gm, "meta")
+        or "compiled_autograd_metrics" not in maybe_gm.meta
+    ):
+        return None
+    return dataclasses.astuple(maybe_gm.meta["compiled_autograd_metrics"])
```

Comment:
Since you created the new CompiledAutogradMetrics, maybe just return it directly instead of converting to a tuple? Makes the usage at the callsite unambiguous at least.